### PR TITLE
add main classes for spark configuration and other bug fix

### PR DESCRIPTION
**Pull request by voidfunction**
The main class can be selected from combo drop list.
untitled

### CODE REVIEW

**Comment by voidfunction** [comment]
Fix #698

**Comment by voidfunction** [comment]
Fix #719 by adding stage id to stage view

**Comment by voidfunction** [comment]
fix #669

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java [hunk 1]
```diff
@@ -59,448 +77,512 @@
...
+		Combo mainClassCombo = new Combo(composite, SWT.DROP_DOWN);
+		mainClassCombo.setLayoutData(gridData);
+		try {
+			java.util.Set<String> classes = getClassesWithMainMethod();
+		    String [] names = new String[classes.size()];
+		    classes.toArray(names);
+		    mainClassCombo.setItems(names);
+		    mainClassCombo.select(0);
+		} catch (CoreException e1) {
+			// Show nothing if we get this exception
```
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Maybe it's better to print the exception out. Just for debugging.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java [hunk 2]
```diff
@@ -59,448 +77,512 @@
...
+			projects = new String[projList.size()];
+			projects = projList.toArray(projects);
+		} catch (Exception e) {
+			PluginUtil.displayErrorDialogAndLog(this.getShell(), "Project Selection Error",
+					"Error occurred while selecting the project.", e);
+		}
+		return projects;
+	}
+
+	private static String javaNatureId = "org.eclipse.jdt.core.javanature";
```
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Could we set this to `final`? I think it's just a constant.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java [hunk 3]
```diff
@@ -59,448 +77,512 @@
...
+					for (ICompilationUnit unit : mypackage.getCompilationUnits()) {
+						IType[] allTypes = unit.getAllTypes();
+						final String unitName = unit.getElementName();
+						for (IType type : allTypes) {
+							IMethod[] methods = type.getMethods();
+							for (IMethod method : methods) {
+								if (method.getElementName().equalsIgnoreCase("main")) {
+
+									// remove .class suffix
+									final String simpleClassName = unitName.substring(0, unitName.length() - 6);
```
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Is all the `unitName` end with `.class`?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java (hunk 2 above)
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Could we set this to final? I think it's just a constant.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java (hunk 1 above)
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Maybe it's better to print the exception out. Just for debugging.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java [hunk 4]
```diff
@@ -152,7 +152,7 @@ protected void execute(@NotNull ExecutionEnvironment environment, @Nullable Call
                                     "Error : Spark batch Job remote debug failed, got exception: " + errorMessage);
 
                             postEventProperty.put("IsSubmitSucceed", "false");
-                            postEventProperty.put("SubmitFailedReason", errorMessage.substring(0, 50));
+                            postEventProperty.put("SubmitFailedReason", errorMessage.substring(0, errorMessage.length() > 50 ?  50 : errorMessage.length()));
```
Comment (ziyel) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java]: Maybe we should use a util class to post the event  and set the max size `50` as a `constant`.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java (hunk 3 above)
Comment (voidfunction) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Yes, those are generated class files by Java or Scala compiler.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java (hunk 1 above)
Comment (voidfunction) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/ui/SparkSubmissionExDialog.java]: Good idea. Fixed. Thx.

**Pull request merged**